packages/carbon/src/abstracts/BaseCommand.ts

```diff
@@ -51,6 +52,12 @@ export abstract class BaseCommand {
 		InteractionContextType.PrivateChannel
 	]
 
+	/**
+	 * The default permission that a user needs to have to use this command.
+	 * This can be overridden by server admins.
+	 */
+	permission?: (typeof Permission)[keyof typeof Permission]
```

Comment:
This type resolves to just `bigint` anyway (so can use bit manipulation), but is any reason this isn't an array that can later be resolved to a bigint?

Reply:
While `keyof typeof Permission` seems to work as expected for me, I was more referring to just 
```suggestion
	permission?: ((typeof Permission)[keyof typeof Permission])[]
```
and then later use reduce in the serialize method to combine those bits
```
this.permissions ? this.permissions.reduce((a, p) => a | p, 0n).toString() : null
```